Kcd/user management super admin view

johanna-skylight opened the pull request:
# PULL REQUEST

## Summary

This PR contains the new `user management` page. It includes the users, user groups and edit members/query sections. All elements are interactive but display static/dummy data. Integration with the server will be made in a second PR.

This PR also includes improvements to the Table component and the Header (dropdown menu) component.

<img width="1512" alt="Screenshot 2025-02-06 at 12 50 51 PM" src="https://github.com/user-attachments/assets/8a480035-dfaa-4f8f-b63f-0a47270c161d" />
<img width="1139" alt="Screenshot 2025-02-06 at 12 50 21 PM" src="https://github.com/user-attachments/assets/12c2dbeb-d92c-4b0a-8356-cde2fa76b2e9" />

## Related Issue

Partially implements [#129](https://linear.app/skylight-cdc/issue/QUE-129/build-user-management-page-view-for-super-admins) and fixes QUE-163

## Additional Information


## Checklist

- [ ] Descriptive Pull Request title
- [ ] Link to relevant issues
- [ ] Provide necessary context for design reviewers
- [ ] Update documentation

fzhao99 commented on query-connector/src/app/ui/designSystem/table/table.module.scss:
```diff
@@ -1,62 +1,62 @@
-    margin: 0 $default-table-padding;
+  thead tr th,
+  tbody tr td {
+    padding-top: $default-table-padding;
+    padding-bottom: $default-table-padding;
+    padding-left: calc($default-table-padding - 10px);
+    padding-right: calc($default-table-padding - 10px);
```
think we can slim these down!
```suggestion
    padding: $default-table-padding calc($default-table-padding - 10px);
```

fzhao99 replied:
generally these changes are much appreciated!

fzhao99 commented on query-connector/src/app/ui/components/header/header.tsx:
```diff
@@ -124,41 +124,51 @@ const HeaderComponent: React.FC<{ authDisabled: boolean }> = ({
+const LOGGED_IN_PATHS = [
+  "/query",
+  "/queryBuilding",
+  "/fhir-servers",
```
Worth a followup PR: might want to swap `fhir-servers` --> `fhirServers` so that these are consistent.

fzhao99 replied:
Ticket here! https://linear.app/skylight-cdc/issue/QUE-163/update-fhir-server-paths-to-fhirserver

fzhao99 commented on query-connector/src/app/(pages)/userManagement/userGroups/page.tsx:
```diff
@@ -0,0 +1,85 @@
+const UserGroups: React.FC = () => {
+  const [userGroups, setUserGroups] = useState<UserGroup[]>([]);
+
+  const { OpenEditSection } = useContext(DataContext);
```
super annoying nit: is there any rule for us using CapitalizedCamelCase like we have here rather than lowerCaseCamelCase? This just jumped out at me readability wise. 

Obviously don't want to spend too much time here, but maybe worth setting up some linting to keep the function names consistently styled

fzhao99 commented on query-connector/src/app/ui/designSystem/table/table.module.scss:
```diff
@@ -1,62 +1,62 @@
-.table {
-  display: flex;
-  flex-direction: column;
+.customizeTableWrapper {
```
Think the table refactor may have broken the styling on some of the other pages

<img width="651" alt="Screenshot 2025-02-06 at 4 29 18 PM" src="https://github.com/user-attachments/assets/b000521c-e65e-4261-ad70-66aab4b55733" />

<img width="1053" alt="Screenshot 2025-02-06 at 4 28 25 PM" src="https://github.com/user-attachments/assets/66007d5e-3c1f-4ea3-a75c-147b8db20694" />

We also may want to extend the component to have a `borderless` variant so that we can match the stylings on the existing result page
<img width="1267" alt="Screenshot 2025-02-06 at 4 29 51 PM" src="https://github.com/user-attachments/assets/0caefbf2-f014-4c2e-8ccf-53b5c8018c29" />

johanna-skylight replied:
I believe I have fixed all the variations of the table in the application, if you still see any issues for sure let me know.

fzhao99 commented on query-connector/src/app/ui/designSystem/table/Table.tsx:
```diff
@@ -15,26 +17,27 @@ type TableProps = {
 const Table: React.FC<TableProps> = ({
   children,
-  bordered = true,
+  bordered = false,
```
Think we'll need to swap the tables in `query-connector/src/app/(pages)/query/components/resultsView/tableComponents` to use the unbordered version.

Before:
<img width="1360" alt="Screenshot 2025-02-06 at 4 32 16 PM" src="https://github.com/user-attachments/assets/85b7331d-7d66-4411-8fbe-9f1f48274564" />

After: 
<img width="1019" alt="Screenshot 2025-02-06 at 4 32 33 PM" src="https://github.com/user-attachments/assets/d9e1d679-293f-45d8-9d48-2f057d6ce2d5" />

fzhao99 commented on query-connector/src/app/(pages)/userManagement/components/DataProvider.tsx:
```diff
@@ -0,0 +1,154 @@
+  },
+};
+
+export const DataContext = createContext<UserManagementContext>({
```
nit: could we rename these `Data*` vars to something more descriptive - (ie `UserContext`?) In a world where we have multiple contexts for multiple things, having more explicit names might make it more clear which context we're using.

johanna-skylight replied:
I have made this change to add the prefix `UserManagement` to all components and objects related to handle the state of the page. I do see that the naming gets verbose. Maybe we should revisit the prefix pattern because one could argue that the context for the different components can be inferred but its location (in this case the UserManagement folder) in the same way that there are multiple page.tsx and layout.tsx files. If multiple components/objects need to be imported with the same name we could use aliases.  

Anyway I added the prefix so will marking this as resolved!

fzhao99 commented on query-connector/src/app/ui/components/header/header.tsx:
```diff
@@ -124,41 +124,51 @@ const HeaderComponent: React.FC<{ authDisabled: boolean }> = ({
+                </li>
+                <li className={styles.subMenuItem}>
+                  <Link className={styles.menuItem} href={"/fhir-servers"}>
+                    FHIR Servers
```
also worth aligning on whether we want Sentence case or Title Case for these. Don't have a strong preference either way (though maybe check with design in case they do?) as long as they're aligned.

johanna-skylight replied:
I have made the change as it was small enough for the `My Queries` menu item.

fzhao99 reviewed (commented):
left a couple of nitpicky comments, but good work overall!

For frontend PR's, we usually drop a slack message in the design channel for Michelle to do a pass once you're ready for her to look at it

johanna-skylight commented:
@fzhao99 I think I addressed all the comments. Thank you for catching all the broken tables 🙇‍♀️  they should be fixed now.

fzhao99 reviewed (approved):
great work on cleaning up the table code!

mikang commented:
Hi @johanna-skylight , looks great! I have a few comments. Could we please make the following updates?

- [ ] Have the dropdown menu (from the gear) be wider, so "User management" is on one line
- [ ] Add toasts when something is changed. "User permissions successfully changed"
- [ ] Change background of the search bar (in the side panel) to be white
- [ ] Move all the content up, so there is a total of 48px of whitespace between the header (including the light blue info header) and the content. This would change the other screens like FHIR server configuration as well.
- [ ] Is it possible to remove the "X" from the dropdown? Otherwise, some users may be "floating" and exist in the app without a user permission status.

<img width="1883" alt="image" src="https://github.com/user-attachments/assets/81306ec9-28df-4497-93a6-ee0558f57796" />

I'm guessing that side panel search and connecting the table rows to the side panel data are not part of this PR - is that right? So, Jane Doe is not in the side panel

I'm not able to exit out of the side panel or change my cursor location when I have the side panel open. I can't navigate when I try tabbing from "2 queries" here (tabbing doesn't do anything). I'll do a broader accessibility sweep later - would you like to make this change now, or later?

<img width="1475" alt="image" src="https://github.com/user-attachments/assets/53a7b383-35da-4f86-9d1b-d18a7de1a087" />

johanna-skylight commented:
@mikang Thanks for the review!

As part of this initial PR I have addressed below items:

- **Have the dropdown menu (from the gear) be wider, so "User management" is on one line.** This is done.
- **Change background of the search bar (in the side panel) to be white.** This has been fixed.
- **Move all the content up, so there is a total of 48px of whitespace between the header (including the light blue info header) and the content. This would change the other screens like FHIR server configuration as well.** I have modified the upper margins in all the pages.
- **Is it possible to remove the "X" from the dropdown? Otherwise, some users may be "floating" and exist in the app without a user permission status.** I have migrated the role dropdown to the same dropdown used in the query page which is a more simpler component that does not have the "X".
 
 **Items that I will not be addressing in this PR.**
 
The toast will get implemented in a follow up PR that integrates the pages with the real data and will allow the user to do the role changes.
 
`Drawer` component accessibility issues. We should create a different ticket to address the component. Because a11y has not been a main focus in the application so far we will probably find that some of the existing components need enhancements. I think it would be a good idea to sync up with Janki to figure out how to treat that as a different workstream from the admin feature.

mikang commented:
All looks and sounds good, @johanna-skylight !! Thank you! Good to go from the design side 👍
